packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,307 @@
+  }
+
+  /**
+   * Checks whether or not visualMedia has user permission
+   *
+   * @beta
+   * @returns Promise that will resolve with true if the user had granted the app permission to media information, or with false otherwise,
+   * In case of an error, promise will reject with the error.
+   * @throws NOT_SUPPORTED_ON_PLATFORM Error if the DevicePermission.Media permission has not successfully granted.
+   */
+  export function hasPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!image.isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+    return sendAndHandleSdkError('permissions.has', permissions);
+  }
+
+  /**
+   * Requests user permission for visualMedia
+   * @beta
+   * @returns Promise that will resolve with true if the user consented permission for media, or with false otherwise,
+   * In case of an error, promise will reject with the error.
+   * @throws NOT_SUPPORTED_ON_PLATFORM Error if the DevicePermission.Media permission has not successfully granted.
+   */
+  export function requestPermission(): Promise<boolean> {
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+    if (!image.isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+    const permissions: DevicePermission = DevicePermission.Media;
+    return sendAndHandleSdkError('permissions.request', permissions);
+  }
+
+  /**
+   * To enable this image capability will let the app developer ask the user to get images from camera/local storage
+   * @beta
+   */
+  export namespace image {
+    const errorInvalidSource = 'Invalid input source.';
+    const errorInvalidCount = 'Invalid input count: Must supply the valid image counts (limit to 10).';
+    /**
+     * Capture one or multiple image(s) using camera.
+     * @beta
+     * @param imageInputs - The input params to customize the image(s) to be captured
+     * @returns Promise that will resolve with {@link VisualMediaFile[]} object or reject with an error.
+     * @throws INVALID_ARGUMENTS Error if imageInputs is null or imageInputs.maxVisualMediaCount is greater than 10 or lesser than 1.
+     */
+    export function captureImages(imageInputs: ImageProperties): Promise<VisualMediaFile[]> {
+      ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      if (!imageInputs || imageInputs.maxVisualMediaCount > 10 || imageInputs.maxVisualMediaCount < 1) {
+        throw errorInvalidCount;
+      }
+      if (imageInputs.sourceProps.source != Source.Camera) {
+        throw errorInvalidSource;
+      }
+      // waiting and return the response from hub-SDK
+      return sendAndHandleSdkError<VisualMediaFile[]>('visualMedia.image.captureImages', imageInputs);
+    }
+
+    /**
+     * Upload the existing image(s) from the gallery.
```

Comment:
Thanks for the naming suggestion!
I will keep the naming captureImages because user can access and passing back the images from gallery during camera flow.
Talked with PM, and we will rename uploadImage in teamsjs codebase with retrieveImages, which is making more sense to the app developers, and keep the uploadImage in hub-sdk. I will make sure I documented why the function naming is different between teamjs and hub-sdk.